internal/pkg/crawler/imdbcrawler.go

```diff
@@ -0,0 +1,160 @@
+package imdb
+
+import (
+	"fmt"
+	"github.com/PhamDuyKhang/littledetective/internal/pkg/flog"
+	"github.com/PuerkitoBio/goquery"
+	"github.com/globalsign/mgo"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type (
+	Film struct {
+		ID          string              `json:"id" bson:"_id"`
+		Rank        int                 `json:"rank" bson:"rank"`
+		URL         string              `json:"url" bson:"url"`
+		Title       string              `json:"title" bson:"title"`
+		Rate        string              `json:"rate" bson:"rate"`
+		ReleaseDate int                 `json:"release_date" bson:"release_date"`
+		Description string              `json:"description" bson:"description"`
+		Credit      map[string][]string `json:"credit" bson:"credit"`
+	}
+)
+
+func Crawler() {
+	l := flog.New()
+	l.SetLocal("imdb")
+	filmIn := make(chan Film, 20)
+	filmOut := make(chan Film, 20)
+	go func() {
+		var wg sync.WaitGroup
+		for i := 1; i <= 20; i++ {
+			wg.Add(1)
+			go ExtractDetail(i, wg, filmIn, filmOut, l)
```

Comment:
You must pass `wg` as the reference here or the function will not work as expected.

Reply:
I don't know why, when I write a sample goroution, I pass the `wg` as value, and the program still right